src/Client/LiveStripeClient.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace MatchBot\Client;
+
+use MatchBot\Domain\Currency;
+use MatchBot\Domain\Money;
+use MatchBot\Domain\StripeCustomerId;
+use Stripe\PaymentIntent;
+use Stripe\StripeClient;
+
+/**
+ * Connects to a real Stripe service, either in test mode or production mode.
+ *
+ */
+class LiveStripeClient implements Stripe
+{
+    public function __construct(private StripeClient $stripeClient)
+    {
+    }
+
+    public function cancelPaymentIntent(string $paymentIntentId): void
+    {
+        $this->stripeClient->paymentIntents->cancel($paymentIntentId);
+    }
+
+    public function updatePaymentIntent(string $paymentIntentId, array $updateData): void
+    {
+        $this->stripeClient->paymentIntents->update($paymentIntentId, $updateData);
+    }
+
+    public function confirmPaymentIntent(string $paymentIntentId): PaymentIntent
+    {
+        return $this->stripeClient->paymentIntents->confirm($paymentIntentId);
+    }
+
+    public function retrievePaymentIntent(string $paymentIntentId): PaymentIntent
```

Comment:
since I've used it four times already here it's probably worth making a type for PaymentIntentID. Either as a class like `StripePaymentIntentId` or something generic like `StripeId<PaymentIntent>`. Would be good to have thoughts on that from @dorota-joanna and @NoelLH